OpenNMT/train.py

```diff
@@ -254,42 +287,47 @@ def main():
 
     print('Building model...')
 
-    if opt.train_from is None:
-        encoder = onmt.Models.Encoder(opt, dicts['src'])
-        decoder = onmt.Models.Decoder(opt, dicts['tgt'])
-        generator = nn.Sequential(
-            nn.Linear(opt.rnn_size, dicts['tgt'].size()),
-            nn.LogSoftmax())
-        if opt.cuda > 1:
-            generator = nn.DataParallel(generator, device_ids=opt.gpus)
-        model = onmt.Models.NMTModel(encoder, decoder, generator)
-        if opt.cuda > 1:
-            model = nn.DataParallel(model, device_ids=opt.gpus)
-        if opt.cuda:
-            model.cuda()
-        else:
-            model.cpu()
-
-        model.generator = generator
+    encoder = onmt.Models.Encoder(opt, dicts['src'])
+    decoder = onmt.Models.Decoder(opt, dicts['tgt'])
 
+    generator = nn.Sequential(
+        nn.Linear(opt.rnn_size, dicts['tgt'].size()),
+        nn.LogSoftmax())
+
+    model = onmt.Models.NMTModel(encoder, decoder)
+
+    if opt.train_from:
```

Comment:
Re: different model options. We could just load in the options from the checkpoint as well to ensure they are the same, though this gets in the way of things like manually lowering learning rates or changing batch sizes or device numbers between checkpoints.